draftjs_exporter/engines/string.py

```diff
@@ -32,54 +32,53 @@ def escape(s):
 }
 
 
+class Elt(object):
+    def __init__(self, type_, attr, children, markup):
+        self.type = type_
+        self.attr = attr
+        self.children = children
+        self.markup = markup
+
+
 class DOMString(DOMEngine):
     """
     String concatenation implementation of the DOM API.
     """
 
     @staticmethod
     def create_tag(type_, attr=None):
-        return {
-            'type': type_,
-            'attr': attr,
-            'children': [],
-        }
+        return Elt(type_, attr, [], None)
 
     @staticmethod
     def parse_html(markup):
         """
         Allows inserting arbitrary HTML into the exporter output.
         Treats the HTML as if it had been escaped and was safe already.
         """
-        return {
-            'type': 'escaped_html',
-            'attr': None,
-            'children': None,
-            'markup': markup,
-        }
+        return Elt('escaped_html', None, None, markup)
 
     @staticmethod
     def append_child(elt, child):
         # This check is necessary because the current wrapper_state implementation
         # has an issue where it inserts elements multiple times.
         # This must be skipped for text, which can be duplicated.
-        is_existing_ref = isinstance(child, dict) and child in elt['children']
+        is_existing_ref = isinstance(child, Elt) and child in elt.children
         if not is_existing_ref:
-            elt['children'].append(child)
+            elt.children.append(child)
 
     @staticmethod
     def render_attrs(attr):
         return ''.join(sorted([' %s="%s"' % (k, escape(v)) for k, v in attr.items()]))
 
     @staticmethod
     def render_children(children):
-        return ''.join([DOMString.render(c) if isinstance(c, dict) else escape(c) for c in children])
+        return ''.join([DOMString.render(c) if isinstance(c, Elt) else escape(c) for c in children])
```

Comment:
Unrelated to the change, but `join` accepts a generator. Therefore instead of passing a list (from the list comprehension), you should be able to pass the generator directly (the comprehension without the square brackets) so the `join` method will iterate the items without the list being fully built in memory.

TL;DR;
Do `return ''.join(DOMString.render(c) if isinstance(c, Elt) else escape(c) for c in children)`

Reply:
I don't see a performance (speed / memory) gain doing this, can't tell whether this is because the optimisation doesn't make big enough of a difference or because my test setup just isn't stable enough.